Formula/rpm.rb

```diff
@@ -143,3 +144,29 @@ def rpmdir(macro)
                          rpmdir("%_rpmdir")/"noarch/test-1.0-1.noarch.rpm"
   end
 end
+
+__END__
+diff --git a/misc/fts.c b/misc/fts.c
+index eadcf5c01..312ed0f2d 100644
+--- a/misc/fts.c
++++ b/misc/fts.c
+@@ -32,7 +32,7 @@ static char sccsid[] = "@(#)fts.c	8.6 (Berkeley) 8/14/94";
+ #endif /* LIBC_SCCS and not lint */
+ 
+ /* Conditional to set up proper fstat64 implementation */
+-#if defined(hpux) || defined(sun) || defined(__APPLE__)
++#if defined(hpux) || defined(sun) || (defined(__APPLE__) && defined(__aarch64__))
```

Comment:
`defined(_DARWIN_FEATURE_ONLY_64_BIT_INODE)` is a nice way to check for this rather than the arch - it's documented in `man 2 stat` (though no one seems to know about it).

Reply:
Yea, I was just reading the docs. Have something else I'm testing that might be nicer to upstream.

Reply:
Ah, looks like your comment came in around the same time I pushed the new patch I'm testing.

Reply:
Hmm no it doesn't work

Reply:
No, I don't think that's right. `_DARWIN_FEATURE_ONLY_64_BIT_INODE` is defined only on ARM macOS (see `sys/cdefs.h`, but `*stat64` is deprecated starting 10.5. The idea is to avoid using `*stat64` directly wherever it is deprecated, but checking for `_DARWIN_FEATURE_ONLY_64_BIT_INODE` only will just remove the use of `*stat64` on ARM macOS but keep it on Intel.

Reply:
Oh I think I misinterpreted your intentions here - I thought your intention was to use stat64 still on Intel. You're unlikely to get very far with defining `_DARWIN_USE_64_BIT_INODE` in the source file since it'll probably be too late - it'll need to be in the build script or something.

But with that said, I'd perhaps check the deployment target. The original patch should only be breaking if the deployment target is <= 10.5. `stat` has been the same as `stat64` for a long time now.

Reply:
It seems trying to do that isn't straightforward, though, since it will probably mess with how the `sys/stat.h` is preprocessed. Will probably just indeed check for `_DARWIN_FEATURE_ONLY_64_BIT_INODE`.

Reply:
Actually, was the only issue a stat64 redefinition? That's not particularly surprising then. It looks like it's only that `#define stat64` that needs wrapping in a `_DARWIN_FEATURE_ONLY_64_BIT_INODE` - this first line here should work fine either way?

Reply:
I didn't look closely at whether it was the only issue, but it was certainly one issue.

The difficulty with checking for `_DARWIN_FEATURE_ONLY_64_BIT_INODE` is that it isn't defined appropriately until we `#include <sys/cdefs.h>`, and that doesn't happen here until the `#include <sys/stat.h>` inside `rpmfts.h`. But if we wait until after then, it's too late.